service/controller/v13/resource/instance/create_terminate_old_workers.go

```diff
@@ -0,0 +1,73 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/azure-operator/service/controller/v13/key"
+	"github.com/giantswarm/azure-operator/service/controller/v13/resource/instance/internal/state"
+	"github.com/giantswarm/microerror"
+)
+
+func (r *Resource) terminateOldWorkersTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding all worker VMSS instances")
+	var allWorkerInstances []compute.VirtualMachineScaleSetVM
+	{
+		allWorkerInstances, err = r.allInstances(ctx, customObject, key.WorkerVMSSName)
+		if IsScaleSetNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find the scale set '%s'", key.WorkerVMSSName(customObject)))
+
+			return currentState, nil
+		} else if err != nil {
+			return "", microerror.Mask(err)
+		}
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d worker VMSS instances", len(allWorkerInstances)))
+
+	c, err := r.getScaleSetsClient(ctx)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "filtering instance IDs for old instances")
+
+	g := key.ResourceGroupName(customObject)
```

Comment:
Do we get charge by character count? **:stuck_out_tongue_winking_eye:**  `resourceGroupName` is more explicit, and makes it easier for me to understand

Reply:
Practices on these tend to vary a bit depending on location, author and age of code. I think we generally lean towards shorter variable names when the scope of use is short or if it's about one specific type that we have documented. Other times then more descriptive ones are used.

I personally vary a bit on these. I'd definitely want to avoid the `superLongExplicitAllInclusiveVariableNamesThatAreUsedInJavaEECulture` but then again, single letter ones are not always good either. The code in here was actually copied / adapted from existing code in `azure-operator`.

Regarding the known variables we also have [fmt entry](https://github.com/giantswarm/fmt/blob/master/go/variable_names.md).

In this case these are just about avoiding long line with embedded function calls on function parameters so I'd like to keep these as-is.